src/java/org/apache/cassandra/db/compaction/UnifiedCompactionStrategy.java

```diff
@@ -1242,7 +1247,7 @@ else if (count <= fanout * controller.getFanout(index + 1) || maxSSTablesToCompa
                     count -= maxSSTablesToCompact;
                 }
 
-                return CompactionAggregate.createUnified(allSSTablesSorted, count, pick, pending, arena, level);
+                return CompactionAggregate.createUnified(allSSTablesSorted, updatedMaxOverlap, pick, pending, arena, level);
```

Comment:
does `UnifiedAggregate#maxOverlap` represent max overlaps for the level or for selected CompactionPick?

Reply:
It represents the max overlap for the level and is important for representing the state of the compaction hierarchy (how late compaction is, what is the extra read amplification from that).

It should be using `maxOverlap`.

`count` may indeed be negative because we don't care that we give a bigger argument to `pullOldestSSTables` than the overlap that actually exists. It should not have been used here.